interfaces/vault_kv/v0/README.md

```diff
@@ -0,0 +1,65 @@
+# `vault-kv`
+
+## Usage
+
+Some charms require a secure key value store. This relation interface describes the expected behavior of any charm claiming to interact with Vault Key Value stores.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Requirer -- mount_suffix, egress_subnet --> Provider
+    Provider -- vault_url, mount, credentials --> Requirer
+```
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+
+Provider expectations
+
+- Must provide the vault url
+- Must provide a key value mount, the mount name shall respect the following pattern: charm-<requirer app>-<user provided suffix>
+- Must provide a role_id and role_secret_id for each unit, with access protected by unit's egress_subnet
+
+### Requirer
+
+Requirer expectations
+
+- Must provide a mount suffix
+- Must provide an egress subnet for each unit used to protect access to the secret backend
+
+## Relation Data
+
+[\[Pydantic Schema\]](./schema.py)
+
+#### Example
+
+```yaml
+provider:
+  app:
+    vault_url: http://10.152.183.104:8200
+    mount: charm-barbican-secrets
+    credentials: |
```

Comment:
Removed the UnitCredentialsSchema, since the secret will encompass both the role_id and the role_secret_id.

```yaml
credentials: |
     {
        "barbican-0": "secret://4f7cc474-a23d-49a2-8b6e-9835c1e08325/cjk5slcrl3uc767oebp0",
        "barbican-1": "secret://4f7cc474-a23d-49a2-8b6e-9835c1e08325/cjk5slcrl3uc767oebpg"
      }
```